felt/core/s3_upload_parameters.py

```diff
@@ -1,69 +1,38 @@
 """
 Felt API s3 upload parameters
 """
-
 from dataclasses import dataclass
-from typing import (
-    Optional,
-    Dict
-)
+from typing import Dict
 
 
 @dataclass
 class S3UploadParameters:
     """
-    Encapsulates parameters for uploading to S3
+    Encapsulates parameters for uploading to S3, including all presigned
+    attributes
     """
 
-    aws_access_key_id: Optional[str]
-    acl: Optional[str]
-    key: Optional[str]
-    policy: Optional[str]
-    signature: Optional[str]
-    success_action_status: Optional[str]
-    x_amz_meta_features_flags: Optional[str]
-    x_amz_meta_file_count: Optional[str]
-    x_amz_security_token: Optional[str]
-    url: Optional[str]
-    layer_id: Optional[str]
-    type: Optional[str]
+    url: str
+    layer_id: str
+    type: str
+    _presigned_attributes: Dict[str, str]
 
     def to_form_fields(self) -> Dict:
         """
-        Returns the form fields required for the upload
+        Returns all form fields including the presigned attributes required
+        for the upload
         """
-        return {
-            'AWSAccessKeyId': self.aws_access_key_id,
-            'key': self.key,
-            'policy': self.policy,
-            'signature': self.signature,
-            'success_action_status': self.success_action_status,
-            'x-amz-meta-feature-flags': self.x_amz_meta_features_flags,
-            'x-amz-meta-file-count': self.x_amz_meta_file_count,
-            'x-amz-security-token': self.x_amz_security_token,
-        }
+        return {**self._presigned_attributes}
```

Comment:
Is this guaranteed to preserve the order of attributes that we received from the server? If the ordering changes then the signature will be invalid.

Reply:
yeah it should preserve order although it would be nice if it were spelled out more explicitly. I added a comment at least.